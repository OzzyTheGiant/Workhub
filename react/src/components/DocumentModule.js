import React from 'react';
import PropTypes from 'prop-types';

class DocumentModule extends React.Component  {
    // {
    //     clients:[
    //         {
    //             clientId:1,
    //             projects:[
    //                 {
    //                     projectId:1,
    //                     documents:[
    //                         {
    //                             documentId:1
    //                         }
    //                     ]
    //                 }
    //             ]
    //         }
    //     ]
    // }

    createList = (clientID, projectID) => {
        let data = this.props.clients;
        // let projects;
        if (clientID) { // get list of projects
            data = data.filter(client => client.id === clientID)[0];
            //if (!data.projects) projects = this.props.getProjects(clientID);
            if (projectID) { // get list of documents
                data = data.filter(project => project.id === projectID);
            }
        }
        this.setState(data)
    }

	render() {
        const clients = this.props.clients;
		return (
            <div id="DocumentModule">
                <ul className="list-view">
                { clients.map((client, index) => <li key={client.id}>{client.clientName}</li>) }
                </ul>
            </div>
        );
	}
}

DocumentModule.propTypes = {
	clients:PropTypes.array,
}

export default DocumentModule;