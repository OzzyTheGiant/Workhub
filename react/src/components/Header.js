import React from 'react';
import PropTypes from 'prop-types';

const Header = (props) => {
    return (
        <header id="Header">
            <img src="http://placehold.it/120x35" alt="Workhub Logo"/>
            <nav>
                <ul>
                    <li onClick={props.logout}>Log Out</li>
                </ul>
            </nav>
        </header>
    );
};

Header.propTypes = {
    logout:PropTypes.func.isRequired
}

export default Header;