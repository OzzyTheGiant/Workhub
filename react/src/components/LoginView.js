import React from 'react';
import PropTypes from 'prop-types';
import Notification from 'components/Notification';

class LoginView extends React.Component {
	constructor() {
		super();
		this.state = {
			username:null,
			password:null,
			errorMessage:null
		}
	};

	onLoginFormSubmit = (e) => {
		if (e) e.preventDefault();
		const data = {username: this.state.username, password: this.state.password};
		this.props.login(data, this.onSuccessfulLogin, this.onFailedLogin);
	}

	onSuccessfulLogin = () => this.props.initApplication();

	onFailedLogin = error => this.setState({
		errorMessage: "Login Error: Username or password is incorrect!"
	});

	render(props) {
		return (
			<div id="LoginView">
				<form action="/login" method="post" onSubmit={this.onLoginFormSubmit}>
					<img src="https://placehold.it/50x50/00638E" alt="Logo"/>
					<h1>Log in to your account</h1>
					<input type="text" name="username" placeholder="Username" onChange={(e) => this.setState({username: e.target.value})}/>
					<input type="password" name="password" placeholder="Password" onChange={(e) => this.setState({password: e.target.value})}/>
					<input type="submit" className="color-button button-ok" value="Log In"/>
				</form>
				<Notification type={this.state.errorMessage ? "error" : ""} message={this.state.errorMessage}/>
			</div>
		);
	}
};

LoginView.propTypes = {
	initApplication:PropTypes.func.isRequired,
	login:PropTypes.func.isRequired
};

export default LoginView;