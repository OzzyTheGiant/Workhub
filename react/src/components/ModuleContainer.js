import React from "react";
import PropTypes from "prop-types";

const ModuleContainer = (props) => {
	return (
		<div id="ModuleContainer">
			<div id="title-bar"><h1>{props.title}</h1></div>
			{props.children}
		</div>
	);
};

ModuleContainer.propTypes = {
	title:PropTypes.string.isRequired
};

export default ModuleContainer;