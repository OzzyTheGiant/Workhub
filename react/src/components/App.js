import React, { Component } from 'react';
import 'css/main-styles.css';
import Header from 'components/Header';
import LoginView from 'components/LoginView';
import DocumentsModule from 'components/DocumentsModule';
import services from 'api/services';
 
class App extends Component {
	constructor() {
		super();
		this.state = {isLoggedIn:false, clients:[]};
	}

	initApplication = () => {
        services.getClients(this.setClients, this.ajaxErrorHandler); 
		this.setState({isLoggedIn:true});
    }

    setClients = (ajaxResponse) => this.setState({
        currentModule:"documents",
        clients:ajaxResponse.data.sort(this.sortClients)
    });

    sortClients = (a, b) => {
        if (a.clientName < b.clientName) return -1;
        if (b.clientName < a.clientName) return 1;
        return 0;
    };

    ajaxErrorHandler = () => {
        // TODO: set error message later
    };

 	render() {
 		const view = this.state.isLoggedIn ? (
			this.state.currentModule === "documents" ? 
            <DocumentsModule clients={this.state.clients} /> : null
		) : (
			<LoginView initApplication={this.initApplication} login={services.login}/>
		);
 		return (
 			<div className="App">
 				{this.state.isLoggedIn ? <Header/> : null}
 				{view}
 				<footer>&copy; 2018 Salinas, Allen & Schmitt, LLP. All Rights Reserved</footer>
 			</div>
 		);
  	}
}

export default App;