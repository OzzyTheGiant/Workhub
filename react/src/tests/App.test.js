import React from 'react';
import { configure, mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import App from 'components/App';
import LoginView from 'components/LoginView';
import Header from 'components/Header';
import ModuleContainer from 'components/ModuleContainer';
import DocumentModule from 'components/DocumentModule';

describe("App", () => {
	/* ==== COMPONENT ==== */
	const app = <App/>;
	let componentWrapper = null;

	beforeAll(() => configure({adapter:new Adapter()}))
	beforeEach(() => componentWrapper = shallow(app))

	it('Should fully render html without crashing', () => {
		render(app);
	});

	it('Should render LoginView on start up', () => {
		const wrapper = mount(app)
		let jsonComponent = toJson(wrapper);
		expect(jsonComponent.type).toBe('App');
		expect(jsonComponent.children[0].type).toBe('div');
		expect(jsonComponent.children[0].children[0].type).toBe('LoginView');
		console.log(jsonComponent);
		expect(jsonComponent).toMatchSnapshot();
	});
});

