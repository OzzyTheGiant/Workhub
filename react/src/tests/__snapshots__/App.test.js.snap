// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render LoginView on start up 1`] = `
<App>
  <div
    className="App"
  >
    <LoginView
      initApplication={[Function]}
      login={[Function]}
    >
      <div
        id="LoginView"
      >
        <form
          action="/login"
          method="post"
          onSubmit={[Function]}
        >
          <img
            alt="Logo"
            src="https://placehold.it/50x50/00638E"
          />
          <h1>
            Log in to your account
          </h1>
          <input
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <input
            className="button button-ok"
            type="submit"
            value="Log In"
          />
        </form>
        <Notification
          message={null}
          type=""
        >
          <div
            className="Notification none"
          />
        </Notification>
      </div>
    </LoginView>
    <footer>
      Â© 2018 Salinas, Allen & Schmitt, LLP. All Rights Reserved
    </footer>
  </div>
</App>
`;
